app/src/main/java/com/github/wnder/guessLocation/GuessFragment.java

```diff
@@ -109,9 +112,16 @@ private void onStyleLoaded(Style style){
         drawCircle(getActivity(), mapboxMap, cameraPosition);
     }
 
-    private void openPreviewActivity() {
+    private void openActivity(Class c) {
         if(networkInfo.isNetworkAvailable()){
-            Intent intent = new Intent(getActivity(), GuessPreviewActivity.class);
+            Intent intent = null;
+            if(c == GuessPreviewActivity.class){
+                intent = new Intent(getActivity(), GuessPreviewActivity.class);
+            }
+            //TODO replace with new activity
+            else if(c == GuessPreviewActivity.class){
+                intent = new Intent(getActivity(), GuessPreviewActivity.class);
+            }
```

Comment:
I understand what you tried to do there, but I think the "Class c" parameter is wrong. You can be more precise on the type of object you want. All our activities extends the AppCompatActivity class, you can use this type instead such that you don't need to check with "if else" statement if the parameter is what we expect or not because it is indeed an activity.

Reply:
I agree that my if-else is not the best idea.
However, I cannot give AppCompatActivity as an argument so I cannot apply your suggestion

Reply:
You can use it like this:

private void openActivity(AppCompatActivity activity) {
        if(networkInfo.isNetworkAvailable()){
            Intent intent = new Intent(getActivity(), activity.getClass());
            startActivity(intent);

and pass a new GuessPreviewActivity as a parameter for the function. Tell me if you find it better or not.

Reply:
Alright, I didn't know that I could create a new activity